### PR TITLE
Track and Set Coordinates of Canvas Objects after they have been dragged

### LINKED ISSUE #10: implement canvas and konva.js to app

### LINKED ISSUE #11: implement functionality to save canvas objects

### PR DESCRIPTION
**Pull request by TomGrzl**
Added event handlers to update object coordinates in the corresponding state via id after they have been dragged

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -16,7 +18,10 @@ function App() {
                 <Switch>
```
Comment (jamarob) [inline frontend/src/App.js:18]: `Router` should go into `index.js`, this allows you to configure different routers for testing. `GlobalStyle.js` can go there too, but that's more a matter of taste.

File: frontend/src/pages/Konva.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import React, {useState} from 'react';
+import {Stage, Layer, Rect, Circle} from 'react-konva';
+
+export default function Konva() {
+    const [rects, setRects] = useState([]);
+    const [circles, setCircles] = useState([])
+    const [rectId, setRectId] = useState(1)
+    const [circleId, setCircleId] = useState(1)
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:8]: This states and the following functions just scream to be refactored into a custom hook :) https://reactjs.org/docs/hooks-custom.html (-> extra ticket, fine for now)

File: frontend/src/pages/Konva.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            stroke: '#000',
+            strokeWidth: 1,
+            shadowBlur: 10
+        }
+        ])
+        setCircleId(circleId + 1);
+    }
+
+
+    const handleDragStart = (e) => {
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:42]: This function does nothing except logging, maybe remove it?

File: frontend/src/pages/Konva.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                        height: 30,
+                        shadowBlur: 10,
+                        fill: '#C9E1F2', shadowOffsetX: 5, shadowOffsetY: 5
+                    }
+                } else {
+                    return {...rect}
+                }
+            }))
+
+        console.log(
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:76]: console.log is super helpful during debugging, but I prefer to not leave it in the code, it has a tendency to litter up the code base and the console.

File: frontend/src/pages/Konva.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                            height={rect.height}
+                            fill={rect.fill}
+                            stroke={rect.stroke}
+                            strokeWidth={rect.strokeWidth}
+                            shadowBlur={rect.shadowBlur}
+                            draggable
+                            isDragging={false}
+                            onDragStart={handleDragStart}
+                            onDragEnd={setRectCoordinatesOnDragEnd}
+                        />))}
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:138]:
```suggestion
                        <Rect
                            {...rect}
                            key={rect.id}
                            draggable
                            isDragging={false}
                            onDragStart={handleDragStart}
                            onDragEnd={setRectCoordinatesOnDragEnd}
                        />))}
```
spread operator is your friend ;)

File: frontend/src/pages/Konva.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+
+    const handleDragStart = (e) => {
+        const id = e.target.id();
+        const x = e.target.x();
+        const y = e.target.y();
+
+        console.log(
+            'ID: ' + id + '  ' +
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:48]: Backtick strings read much nicer imho

File: frontend/src/pages/Konva.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                        x: x,
+                        y: y,
+                        isDragging: false,
+                        width: 30,
+                        height: 30,
+                        shadowBlur: 10,
+                        fill: '#C9E1F2', shadowOffsetX: 5, shadowOffsetY: 5
+                    }
+                } else {
+                    return {...rect}
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:72]: Simply return `rect`? I think there is no need to create a new object, especially if you do this on every drag end for every rect (and circle)

File: frontend/src/pages/Konva.js [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    const logCircles = () => {
+        console.log(circles)
+    }
+
+    return (
+        <>
+            <button onClick={addRect}>Add Rect</button>
+            <button onClick={addCircle}>Add Circle</button>
+            <button onClick={logRects}>Log Rects</button>
+            <button onClick={logCircles}>Log Circles</button>
```
Comment (jamarob) [inline frontend/src/pages/Konva.js:119]: Nice for experimenting.. but I don't think seeing logs is part of your user story ;)

**Review by jamarob — CHANGES_REQUESTED** [review]
Coolio 😎 ! Easy to review!

Take a look at your onDragEnd functions, that unnecessary object creation should be changed. 

The Konva component is getting really big, I think there is some potential for refactoring.

**Comment by TomGrzl** [comment]
Thanks for the feedback, it will be applied with the next pull request :)

**Pull request merged**